modules/components/core/org.wso2.carbon.pc.core/src/main/java/org/wso2/carbon/pc/core/ProcessStore.java

```diff
@@ -917,17 +947,149 @@ public String addPredecessor(String predecessorDetails){
                 if(predecessorList.length() != 0){
                     Element rootElement = doc.getDocumentElement();
                     for(int i = 0 ; i < predecessorList.length() ; i++){
-                        Element successorElement = append(doc, rootElement, "predecessor", mns);
-                        appendText(doc, successorElement, "name", mns, predecessorList.getJSONObject(i).getString("name"));
-                        appendText(doc, successorElement, "path", mns, predecessorList.getJSONObject(i).getString("path"));
-                        appendText(doc, successorElement, "id", mns, predecessorList.getJSONObject(i).getString("id"));
+                        Element predecessorElement = append(doc, rootElement, "predecessor", mns);
+                        appendText(doc, predecessorElement, "name", mns, predecessorList.getJSONObject(i).getString("name"));
+                        appendText(doc, predecessorElement, "path", mns, predecessorList.getJSONObject(i).getString("path"));
+                        appendText(doc, predecessorElement, "id", mns, predecessorList.getJSONObject(i).getString("id"));
+                    }
+                    String newProcessContent = xmlToString(doc);
+                    resource.setContent(newProcessContent);
+                    reg.put(processAssetPath, resource);
+                }
+            }
+
+        }catch (Exception e){
+            log.error(e);
+        }
+        return "OK";
+    }
```

Comment:
Because here I used a jaggery file to invoke to the backend code which provided in the ProcessStore and print the response as a String. Now I changed it to the boolean type and cahnge the jag files also. fixed in https://github.com/wso2/product-pc/pull/36